arwenmandos/execScenario.go

```diff
@@ -17,18 +17,21 @@ func (ae *ArwenTestExecutor) Reset() {
 func (ae *ArwenTestExecutor) ExecuteScenario(scenario *mj.Scenario, fileResolver fr.FileResolver) error {
 	ae.fileResolver = fileResolver
 	ae.checkGas = scenario.CheckGas
+	resetGasTracesIfNewTest(ae, scenario)
+
 	err := ae.InitVM(scenario.GasSchedule)
 	if err != nil {
 		return err
 	}
 
 	txIndex := 0
 	for _, generalStep := range scenario.Steps {
+		enableGasTraceInRuntime(ae)
```

Comment:
instead of enable and disable you could have one function only Set - which receives a bool. Simplifies this a lot.